packages/open-payments/src/client/requests.ts

```diff
@@ -65,12 +65,36 @@ export const get = async <T>(
 
 export const createAxiosInstance = (args: {
   requestTimeoutMs: number
+  privateKey: KeyLike
+  keyId: string
 }): AxiosInstance => {
   const axiosInstance = axios.create({
     timeout: args.requestTimeoutMs
   })
-
   axiosInstance.defaults.headers.common['Content-Type'] = 'application/json'
 
+  axiosInstance.interceptors.request.use(
+    async (config) => {
+      const sigHeaders = await createSignatureHeaders({
+        request: {
+          method: config.method.toUpperCase(),
+          url: config.url,
+          // https://github.com/axios/axios/issues/5089#issuecomment-1297761617
+          headers: JSON.parse(JSON.stringify(config.headers)),
+          body: config.data
+        },
+        privateKey: args.privateKey,
+        keyId: args.keyId
+      })
+      config.headers['Signature'] = sigHeaders['Signature']
+      config.headers['Signature-Input'] = sigHeaders['Signature-Input']
+      return config
+    },
+    null,
+    {
+      runWhen: (config) => !!config.headers['Authorization']
```

Comment:
Because we don't need `Authorization` header on the AS calls, this would effectively handle both AS and RS with no problem, correct?

Reply:
I think there _are_ some AS calls that require the `Authorization` header
https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-7.3.1.1

I was actually thinking that the grant initiation request might be the only one with it, but I'm not sure

Reply:
Does that mean that the auth server open payment specs are missing the `signature-input` and `signature` header parameters in the open api spec files? If they need to provide an `Authorization` token?

Reply:
Yeah, looks like that's the case.
I just opened:
- https://github.com/interledger/open-payments/issues/209

Reply:
@wilsonianb for my own understanding, wouldn't grant initiation be the only AS request _without_ the `Authorization: GNAP {accessToken}` header?

Reply:
yeah, I think that's correct